cpp/src/command_classes/CentralScene.cpp

```diff
@@ -201,10 +234,52 @@ bool CentralScene::HandleMsg
 
 		if( Node* node = GetNodeUnsafe() )
 		{
-				char lbl[64];
-				for (int i = 1; i <= m_scenecount; i++) {
-					snprintf(lbl, 64, "Scene %d", i);
-					node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), _instance, i, lbl, "", true, false, 0, 0 );
+				for (int i = 1; i <= m_scenecount && i <= 31; i++) {
+					if ( version2orHigher )
+					{
+					    if ( identical )
+					    {
+					        // instance for scene number normaly isn't right, but in this case i think this is the best solution.
+					        // Because i higly doubt that someone will ever use the multichannel class together with the centralScene class.
+                            int keyAttributes = _data[3];
+                            if ( keyAttributes & CentralSceneMask_KeyPressed1time)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_SinglePress, "Scene", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed2times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_DoublePress, "Scene double press", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_HeldDown)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_HeldDown, "Scene held down", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyReleased)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Released, "Scene released", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed3times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_TriplePress, "Scene triple press", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed4times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Pressed4Times, "Scene pressed 4 times", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed5times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Pressed5Times, "Scene pressed 5 times", "", true, false, 0, 0 );
+                            }
+                        }
```

Comment:
I also put some thoughts into this:
Looking at the spec, it says that there can be only 3 scenes with different keyAttributes (Number of Bit Mask Bytes is just 2 Bit) so we could use that.
Have one ValueID like "Scenes identical" with value 0 and 1.
Then in case 1: use the Filterlist you suggested.
And in case 0: have 1 to 3 Filterlists with the Key Attributes for each scene. 

So in conclusion which indexes this solution will have:
index 0: Scene Count
index 1: Scene KeyAttribute
index 2: Scene Number
index 3: Button (not sure why we need this one?)
index 4: Scene Identical
index 5: Supported Key Attributes All Scene (only if all are identical)
index 6: Supported Key Attributes Scene 1
index 7: Supported Key Attributes Scene 2
index 8: Supported Key Attributes Scene 3
index 5+n: Supported Key Attributes Scene n

If Sigma decides to use the remaining Reserved bits (3-6) for more Scene Numbers we are still save because the maximum for this is 2^6= 64 (2 bit in this byte already taken, and I don't think they will change this for compatibility reason).

What do you think about this?